views/install.sh.erb

```diff
@@ -121,9 +119,6 @@ elif test -f "/etc/redhat-release"; then
   if test "$platform" = "xenserver"; then
     # Current XenServer 6.2 is based on CentOS 5, platform is not reset to "el" server should hanlde response
     platform="xenserver"
-  else
-    # FIXME: use "redhat"
-    platform="el"
```

Comment:
Okay I think you're not being aggressive enough about removing this code then, and yet also not quite being careful enough.

The test above this to set platform to "xenserver" if platform is "xenserver" can be dropped since its not useful anymore.  Also the if "fedora" should be dropped since the platform version remapping is done in platform.rb

The problem is that we don't have good testing around this at all, particularly for the edge cases, and you've dropped the else which may catch all kinds of strange edge conditions.  What probably needs to happen is that we need to look for the strings which platform.rb does support ("xenserver", "fedora", "redhat", etc) and then pass those unmodified, but then still maintain the fall through so that if we hit a string here that platform.rb does not support then we still remap to "el".

After deploying this if we check the logs and see that "el" disappears then we know we can drop the else statement, otherwise we need to figure a way to audit what is still hitting it.